app/ai/voice/agents/breeze_buddy/template/loader.py

```
@@ -275,3 +312,59 @@ async def load_template(
+        spreadsheet_id = extract_spreadsheet_id(ref.spreadsheet_url)
+        if not spreadsheet_id:
+            logger.warning(
+                "Data source '%s' has malformed spreadsheet_url: %s",
+                ref.name,
+                ref.spreadsheet_url,
+            )
+            return DATA_SOURCE_UNAVAILABLE
+
+        sheet = ref.sheet_name or "_first_"
+        cols = json.dumps(sorted(ref.columns or []))
+        cols_digest = hashlib.md5(cols.encode()).hexdigest()[:8]
```

Ruff (0.15.17)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)